dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -0,0 +1,317 @@
+                    .search(GET_ACTIVE_VANITY_URL + " +conHost:" + hostId
+                            + GET_VANITY_URL_LANGUAGE_ID
+                            + languageId, 0, 0, StringPool.BLANK, user, false);
+            contentResults.stream().forEach((Contentlet con) -> {
+                VanityUrl vanityUrl = getVanityUrlFromContentlet(con);
+                addToVanityURLCache(vanityUrl);
+                results.add(vanityUrl);
+            });
+        } catch (DotDataException | DotSecurityException e) {
+            Logger.error(this, "Error searching for active Vanity URLs", e);
+        }
+        return results.build();
+    }
+
+    @Override
+    public List<CachedVanityUrl> getActiveCachedVanityUrls(final User user) {
+        ImmutableList.Builder<CachedVanityUrl> results = new ImmutableList.Builder();
+        List<VanityUrl> vanityUrls = getActiveVanityUrls(user);
+        vanityUrls.stream().forEach((VanityUrl vanityUrl) -> {
+            addToVanityURLCache(vanityUrl);
+            results.add(new CachedVanityUrl(vanityUrl));
+        });
+        return results.build();
+    }
+
+    @Override
+    public VanityUrl getVanityUrlFromContentlet(final Contentlet con) {
+        if (con != null) {
+            try {
+                if (!con.isVanityUrl()) {
+                    throw new DotStateException(
+                            "Contentlet : " + con.getInode() + " is not a Vanity Url");
+                }
+            } catch (DotDataException | DotSecurityException e) {
+                throw new DotStateException(
+                        "Contentlet : " + con.getInode() + " is not a Vanity Url", e);
+            }
+        } else {
+            throw new DotStateException("Contentlet is null");
+        }
+
+        DefaultVanityUrl vanityUrl = new DefaultVanityUrl();
+        vanityUrl.setStructureInode(con.getContentTypeId());
+        try {
+            contentletAPI.copyProperties((Contentlet) vanityUrl, con.getMap());
+        } catch (Exception e) {
+            throw new DotStateException("Vanity Url Copy Failed", e);
+        }
+        vanityUrl.setHost(con.getHost());
+        if (UtilMethods.isSet(con.getFolder())) {
+            try {
+                Folder folder = APILocator.getFolderAPI()
+                        .find(con.getFolder(), APILocator.systemUser(), false);
+                vanityUrl.setFolder(folder.getInode());
+            } catch (Exception e) {
+                vanityUrl = new DefaultVanityUrl();
+                Logger.warn(this, "Unable to convert contentlet to Vanity Url " + con, e);
+            }
+        }
+
+        return vanityUrl;
+    }
+
+    /**
+     * Return the VanityUrl object from the given contentlet
+     * and update the value on cache
+     *
+     * @param contentlet Vanity Url Contentlet
+     * @return the VanityUrl object
+     */
+    private VanityUrl transformContentletToVanityUrl(Contentlet contentlet) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "transformContentletToVanityUrl" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)